test/e2e/tests/confirmations/signatures/sign-typed-data-v4.spec.ts

```diff
@@ -0,0 +1,107 @@
+        //
+        // await driver.clickElement('.confirm-scroll-to-bottom__button');
+        // await driver.clickElement('[data-testid="confirm-footer-button"]');
+
+        await assertVerifiedResults(driver, publicAddress);
+      },
+    );
+  });
+
+  it('initiates and rejects', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver, ganacheServer }: { driver: Driver, ganacheServer: Ganache }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#signTypedDataV4');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog, { timeout: 20000 });
+        await driver.clickElement('[data-testid="confirm-footer-cancel-button"]');
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const rejectionResult = await driver.waitForSelector({
+          css: '#signTypedDataV4Result',
+          text: 'Error: User rejected the request.',
+        });
+        assert.ok(rejectionResult);
+      },
+    );
+  });
+});
+
+async function assertInfoValues(driver: Driver) {
+  const origin = driver.findElement({ text: DAPP_HOST_ADDRESS });
+  const contractPetName = driver.findElement({
+    css: '.name__value',
+    text: '0xCcCCc...ccccC',
+  });
+
+  const primaryType = driver.findElement({ text: 'Mail' });
+  const contents = driver.findElement({ text: 'Hello, Bob!' });
+  const fromName = driver.findElement({ text: 'Cow' });
+  const fromAddressNum0 = driver.findElement({ css: '.name__value', text: '0xCD2a3...DD826' });
+  const toName = driver.findElement({ text: 'Bob' });
+  const toAddressNum2 = driver.findElement({ css: '.name__value', text: '0xB0B0b...00000' });
+  const attachment = driver.findElement({ text: '0x' });
+
+  assert.ok(await origin, 'origin');
+  assert.ok(await contractPetName, 'contractPetName');
+  assert.ok(await primaryType, 'primaryType');
+  assert.ok(await contents, 'contents');
+  assert.ok(await fromName, 'fromName');
+  assert.ok(await fromAddressNum0, 'fromAddressNum0');
+  assert.ok(await toName, 'toName');
+  assert.ok(await toAddressNum2, 'toAddressNum2');
+  assert.ok(await attachment, 'attachment');
+}
+
+async function assertVerifiedResults(driver: Driver, publicAddress: string) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement('#signTypedDataV4Verify');
+
+  const verifyResult = await driver.findElement('#signTypedDataV4Result');
+  await driver.waitForSelector({
+    css: '#signTypedDataV4VerifyResult',
+    text: publicAddress,
+  });
+  const verifyRecoverAddress = await driver.findElement('#signTypedDataV4VerifyResult');
+
+  assert.equal(await verifyResult.getText(), '0xcd2f9c55840f5e1bcf61812e93c1932485b524ca673b36355482a4fbdf52f692684f92b4f4ab6f6c8572dacce46bd107da154be1c06939b855ecce57a1616ba71b');
+  assert.equal(await verifyRecoverAddress.getText(), publicAddress);
+}
```

Comment:
the confirmation-redesign test builds seem to be flaky 🤔  
![CleanShot 2024-05-09 at 21 55 31@2x](https://github.com/MetaMask/metamask-extension/assets/20778143/697bb83e-eb46-40c4-896c-5e2e2e6a1085)

Reply:
this is because sometimes there is a delay in the dialog opening up. switch  `await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);` to the method  `await switchToNotificationWindow(driver);` This should make it less flaky 🤞

Reply:
ohh I see now. Thank you @pnarayanaswamy! updated [bd82ec6](https://github.com/MetaMask/metamask-extension/pull/24352/commits/bd82ec636ddb44f24eefc41f0dd145ba8a887526)